torch_xla/csrc/tensor.cpp

```diff
@@ -79,6 +79,14 @@ XLATensorPtr XLATensor::Create(
     c10::optional<at::ScalarType> logical_element_type) {
   XLATensorPtr xtensor = c10::make_intrusive<XLATensor>(
       XLATensor(std::move(ir_value), device, logical_element_type));
+  // Preserve sharding if a new tensor is created from a sharded IR node.
+  if (ir_value) {
```

Comment:
1. `ir_value` is already std::moved, I think it will always be null?
2. why not do this logic in `XLATensor`'s constrcutor? `Create` message suppose to just register the created tensor.

Reply:
Good catch, and yes should do this using the constructor now!

Reply:
Done.